### PR TITLE
Certik audit fix

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Resolves the following Certik report fixes:
    • VEN-970 RFR-02 | Duplicate Imports
    • VEN-966 VTV-05 | Inefficient Memory/Storage Parameters
    • VEN-967 CVP-05 | Storage Variables Declared Outside of `ComptrollerV1Storage`
    • VEN-973 VPB-13 | Typos and Inconsistencies
    • VEN-974 VTV-04 | Unnecessary Reference to `VTokenInterface`
    • VEN-963 RDR-02 | Missing/Unnecessary Functionality
    • VEN-964 VPB-14 | Unused State Variable

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

**Review by chechu — COMMENTED** [review]
I miss the second recommendation in VEN-966: using calldata instead of memory for the functions mentioned in AccessControlManager.sol.

File: contracts/Comptroller.sol [hunk 1]
```diff
@@ -15,38 +15,6 @@ import "./Governance/AccessControlManager.sol";
...
-
-    // closeFactorMantissa must not exceed this value
-    uint256 internal constant closeFactorMaxMantissa = 0.9e18; // 0.9
-
-    // No collateralFactorMantissa may exceed this value
-    uint256 internal constant collateralFactorMaxMantissa = 0.9e18; // 0.9
-
     // PoolRegistry, immutable to save on gas
     /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
     address public immutable poolRegistry;
```
Comment (chechu) [inline contracts/Comptroller.sol:20]: According to VEN-967, shouldn't we move also these variables (the rest ones: poolRegistry, accessControl, rewardsDistributors, rewardsDistributorExists) to the ComptrollerV1Storage contract

File: contracts/ComptrollerStorage.sol [hunk 2]
```diff
@@ -20,11 +20,6 @@ contract ComptrollerV1Storage {
      */
     uint256 public liquidationIncentiveMantissa;
 
-    /**
-     * @notice Max number of assets a single account can participate in (borrow or use as collateral)
-     */
-    uint256 public maxAssets;
-
     /**
      * @notice Per-account mapping of "assets you are in", capped by maxAssets
```
Comment (chechu) [inline contracts/ComptrollerStorage.sol:24]:
```suggestion
     * @notice Per-account mapping of "assets you are in"
```

File: contracts/Comptroller.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Comptroller.sol:20]: Hey @chechu ,

The rest ones you mentioned are immutable and private variables which can't be moved to the base contract `ComptrollerV1Storage`.

**Comment by defcon022** [comment]
> I miss the second recommendation in VEN-966: using calldata instead of memory for the functions mentioned in AccessControlManager.sol.

Done - https://github.com/VenusProtocol/isolated-pools/pull/151/commits/39a1a17c0af2d0c15fc83fb7c88d8a346e450f41

**Review by chechu — APPROVED** [review]

**Review by narayanprusty — APPROVED** [review]

File: contracts/Rewards/RewardsDistributor.sol [hunk 3]
```diff
@@ -405,32 +401,21 @@ contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable {
      * @notice Claim all rewardToken accrued by the holders
```
Comment (0xlucian) [inline contracts/Rewards/RewardsDistributor.sol:401]:
```suggestion
     * @notice Claim all rewardToken accrued by the holders.
```

File: contracts/Pool/PoolRegistry.sol [hunk 4]
```diff
@@ -320,7 +319,7 @@ contract PoolRegistry is Ownable2StepUpgradeable {
     }
 
     /**
-     * @param comptroller The Comptroller implementation address.
+     * @param comptroller The comptroller proxy address associated to the pool.
      * @notice Returns Venus pool.
```
Comment (0xlucian) [inline contracts/Pool/PoolRegistry.sol:323]: use @return tag for return values?

**Review by 0xlucian — COMMENTED** [review]
Overall seems good. Only left comments for natspec fixes. Trying to be consistent trhoughout our codebase. AFAIK as of now  `@dev` and `@notice` should end with a full-stop, whereas `@param` and  `@return` should not.

File: contracts/ComptrollerStorage.sol [hunk 5]
```diff
@@ -94,4 +72,36 @@ contract ComptrollerV1Storage {
...
+        uint256 effects;
+        uint256 liquidity;
+        uint256 shortfall;
+    }
+
+    struct RewardSpeeds {
+        address rewardToken;
+        uint256 supplySpeed;
+        uint256 borrowSpeed;
+    }
```
Comment (kkirka) [inline contracts/ComptrollerStorage.sol:95]: Let's move these to the beginning of the contract before state variable declarations
https://docs.soliditylang.org/en/v0.8.17/style-guide.html#order-of-layout

File: contracts/RiskFund/ProtocolShareReserve.sol [hunk 6]
```diff
@@ -23,11 +23,11 @@ contract ProtocolShareReserve is Ownable2StepUpgradeable, ExponentialNoError, Re
 
     /**
      * @dev Initializes the deployer to owner.
-     * @param _protocolIncome  remaining protocol income.
+     * @param _protocolIncome The address, protocol income will be sent to.
```
Comment (kkirka) [inline contracts/RiskFund/ProtocolShareReserve.sol:26]:
```suggestion
     * @param _protocolIncome The address protocol income will be sent to.
```

File: contracts/Pool/PoolRegistry.sol (hunk 4 above)
Comment (defcon022) [reply contracts/Pool/PoolRegistry.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/Pool/PoolRegistry.sol [hunk 7]
```diff
@@ -320,7 +319,7 @@ contract PoolRegistry is Ownable2StepUpgradeable {
     }
 
     /**
-     * @param comptroller The Comptroller implementation address.
+     * @param comptroller The comptroller proxy address associated to the pool.
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Pool/PoolRegistry.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/Rewards/RewardsDistributor.sol [hunk 8]
```diff
@@ -221,11 +216,12 @@ contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable {
         _distributeSupplierRewardToken(vToken, supplier);
     }
 
+    /**
+     * @notice Calculate REWARD TOKEN accrued by a supplier and possibly transfer it to them
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Rewards/RewardsDistributor.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/Rewards/RewardsDistributor.sol [hunk 9]
```diff
@@ -258,7 +254,7 @@ contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable {
      * @notice Calculate reward token accrued by a borrower and possibly transfer it to them
      *         Borrowers will begin to accrue after the first interaction with the protocol.
      * @dev This function should only be called when the user has a borrow position in the market
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Rewards/RewardsDistributor.sol:267]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/Rewards/RewardsDistributor.sol (hunk 3 above)
Comment (defcon022) [reply contracts/Rewards/RewardsDistributor.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/Comptroller.sol [hunk 10]
```diff
@@ -1207,7 +1153,7 @@ contract Comptroller is Ownable2StepUpgradeable, ComptrollerV1Storage, Comptroll
     }
 
     /**
-     * @dev Internal function to check that vTokens can be safelly redeemed for the underlying asset
+     * @dev Internal function to check that vTokens can be safely redeemed for the underlying asset
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Comptroller.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/ComptrollerStorage.sol (hunk 5 above)
Comment (defcon022) [reply contracts/ComptrollerStorage.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

File: contracts/RiskFund/ProtocolShareReserve.sol (hunk 6 above)
Comment (defcon022) [reply contracts/RiskFund/ProtocolShareReserve.sol]: Fixed , can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/151/commits/ffd493f85fd85b0e1ede68a5b5b8ca0e298698e4).

**Review by narayanprusty — APPROVED** [review]

**Review by 0xlucian — APPROVED** [review]
lgtm

**Pull request merged**